float64 current_x
---
float64 error_x
float64 error_y